File: cpp/se/AuDomainObjectType.cpp
#include "AuDomainObjectType.hpp"

#include <string>

const std::string& AuDomainObjectType::getName() const {
    static const std::string name = "audom";
    return name;
}

const std::string& AuDomainObjectType::getURI() const {
    static const std::string uri = "urn:X-au:params:xml:ns:audomain-1.0";
    return uri;
}

const std::string& AuDomainObjectType::getSchemaLocation() const {
    static const std::string schemaLocation =
            "urn:X-au:params:xml:ns:audomain-1.0 audomain-1.0.xsd";
    return schemaLocation;
}

const std::string& AuDomainObjectType::getIdentType() const {
    static const std::string ident = "name";
    return ident;
}

File: src/main/java/com/ausregistry/jtoolkit2/se/AuDomainObjectType.java
package com.ausregistry.jtoolkit2.se;

/**
 * Represents the audomain-1.0 EPP extension object.
 * @deprecated Should instead use {@link ExtendedObjectType#AU_DOMAIN}.
 */
class AuDomainObjectType implements ObjectType {
    private static final long serialVersionUID = -543331887000261081L;

    public String getName() {
        return "audom";
    }

    public String getURI() {
        return "urn:X-au:params:xml:ns:audomain-1.0";
    }

    public String getSchemaLocation() {
        return "urn:X-au:params:xml:ns:audomain-1.0 audomain-1.0.xsd";
    }

    public String getIdentType() {
        return "name";
    }
}

